### PR TITLE
SALTO-5964-pager-duty-adapter-mvp

### PR DESCRIPTION
**Pull request by Uri-Bar**
SALTO-5964-pager-duty-adapter-mvp
---
_Release Notes_: 
pagerDuty: release of the MVP adapter

---
_User Notifications_: 
none

### CODE REVIEW

File: packages/adapter-components/test/fetch/request/pagination/pagination_functions.test.ts [hunk 1]
```diff
@@ -98,6 +99,26 @@ describe('pagination functions', () => {
     })
   })
 
+  describe('offsetAndLimitPagination', () => {
+    it('should calculate next pages', async () => {
+      const paginate = offsetAndLimitPagination()
+      expect(
```
Comment (omrifarkash) [inline packages/adapter-components/test/fetch/request/pagination/pagination_functions.test.ts:105]: Suggestion:
Maybe split into two `it` blocks one for each `expect` assertion

File: packages/pager-duty-adapter/src/definitions/references.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  {
+    src: { field: 'id', parentTypes: ['escalationPolicy__escalation_rules__targets'] },
+    serializationStrategy: 'id',
+    target: { type: 'schedule' },
+  },
+  // {
+  //   src: { field: 'escalation_policy', parentTypes: ['service'] },
+  //   serializationStrategy: 'escalationPolicy',
+  //   target: { type: 'escalationPolicy' },
+  // },
```
Comment (omrifarkash) [inline packages/pager-duty-adapter/src/definitions/references.ts:39]: leftover

File: packages/pager-duty-adapter/src/definitions/references.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    target: { type: 'service' },
+  },
+]
+
+export const REFERENCES: definitions.ApiDefinitions<Options>['references'] = {
+  rules: REFERENCE_RULES,
+  contextStrategyLookup: {
+    parentType: ({ instance }) => _.get(instance.value, 'parent_type'),
+  },
+  // I did not use it in the end but the reference type must have serializationStrategyLookup defined
```
Comment (omrifarkash) [inline packages/pager-duty-adapter/src/definitions/references.ts:73]: Didn't understand the comment, can you elaborate?

File: packages/pager-duty-adapter/src/definitions/references.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import _ from 'lodash'
+import { Value } from '@salto-io/adapter-api'
+import { definitions, references as referenceUtils } from '@salto-io/adapter-components'
+import { ReferenceContextStrategies, Options, CustomReferenceSerializationStrategyName } from './types'
```
Comment (omrifarkash) [inline packages/pager-duty-adapter/src/definitions/references.ts:19]: Please use type name constants in this file

File: packages/pager-duty-adapter/src/definitions/requests/pagination.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { definitions, fetch as fetchUtils } from '@salto-io/adapter-components'
+import { ClientOptions, PaginationOptions } from '../types'
+
+const { offsetAndLimitPagination } = fetchUtils.request.pagination
+
+export const PAGINATION: Record<PaginationOptions, definitions.PaginationDefinitions<ClientOptions>> = {
+  cursor: {
```
Comment (omrifarkash) [inline packages/pager-duty-adapter/src/definitions/requests/pagination.ts:22]: Is `offsetAndLimitPagination` a cursor pagination method?

File: packages/pager-duty-adapter/test/fetch_mock_replies.json [hunk 6]
```diff
@@ -0,0 +1,325 @@
+[
```
Comment (omrifarkash) [inline packages/pager-duty-adapter/test/fetch_mock_replies.json:1]: Please make this one liner

**Review by omrifarkash — COMMENTED** [review]
Nice!
leaving the approval to @netama

File: packages/pager-duty-adapter/src/client/connection.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  } catch (e) {
+    log.error('Failed to validate credentials: %s', e)
+    throw new clientUtils.UnauthorizedError(e)
+  }
+}
+
+export const createConnection: clientUtils.ConnectionCreator<Credentials> = retryOptions =>
+  clientUtils.axiosConnection({
+    retryOptions,
+    baseURLFunc: async () => 'https://api.pagerduty.com',
```
[the remark this answers is not in the document]
Comment (Uri-Bar) [reply packages/pager-duty-adapter/src/client/connection.ts:42]: the subdomain is in different endpoint than the API. this endpoint is generic

File: packages/pager-duty-adapter/src/definitions/fetch/fetch.ts [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+  },
+  updated_at: {
+    omit: true,
+  },
+  created_by: {
+    omit: true,
+  },
+  updated_by: {
+    omit: true,
+  },
```
[the remark this answers is not in the document]
Comment (Uri-Bar) [reply packages/pager-duty-adapter/src/definitions/fetch/fetch.ts:43]: I saw, but why do we need them? if there is no use for them at all I dont see point in hiding them.
off course we could easly change it in the future if we need

File: packages/adapter-components/test/fetch/request/pagination/pagination_functions.test.ts (hunk 1 above)
Comment (Uri-Bar) [reply packages/adapter-components/test/fetch/request/pagination/pagination_functions.test.ts:105]: just copied to format of all the other test, it seems ok to me as it checks one flow of pagination

File: packages/pager-duty-adapter/test/fetch_mock_replies.json (hunk 6 above)
Comment (Uri-Bar) [reply packages/pager-duty-adapter/test/fetch_mock_replies.json:1]: talked f2f

File: packages/pager-duty-adapter/src/definitions/requests/pagination.ts (hunk 5 above)
Comment (Uri-Bar) [reply packages/pager-duty-adapter/src/definitions/requests/pagination.ts:22]: fixed

**Review by omrifarkash — APPROVED** [review]

**Pull request merged**